src/sagemaker/image_uris.py

```diff
@@ -154,6 +154,18 @@ def retrieve(
             )
 
     if is_jumpstart_model_input(model_id, model_version):
+        if non_none_fields := {
+            key: value
+            for key, value in args.items()
+            if key in {"version", "framework", "container_version", "py_version"}
+            and value is not None
+        }:
+            JUMPSTART_LOGGER.info(
```

Comment:
thanks - I would vote for a `warning` but your call :) 

technically, this is still backward incompatible, but the risk of breaking any customers is really low (it would require cx to have written code than calls the `retrieve` method w/ both a `model_id` and a `framework`, `container_version` and/or `py_version` which is unlikely.